File: 0383-ransom-note/0383-ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        Dictionary<char,int> dic = new Dictionary<char,int>();
        foreach(var c in magazine){
            if(!dic.ContainsKey(c)){
                dic.Add(c,1);
            }else{
                dic[c]++;
            }
        }
        foreach(var c in ransomNote){
            if(dic.ContainsKey(c) && dic[c] > 0){
                dic[c]--;
            }else{
                return false;
            }
        }
        return true;
    }
}

File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> mp;
        for(auto c : magazine){
            if(mp.find(c) != mp.end()){
                mp[c]++;
            }else{
                mp.insert({c,1});
            }
        }
        for(auto c : ransomNote){
            if(mp.find(c) != mp.end()){
                if(mp[c] > 0){
                    mp[c]--;
                }else{
                    return false;
                }
            }else{
                return false;
            }
        }

        return true;
    }
};